### PR TITLE
Issue438 records create

### PR DESCRIPTION
**Pull request by jknnr**

### CODE REVIEW

**Comment by maxvisser** [comment]
On Pairing screen; profile icon is on back button
image

File: presentation/src/main/res/layout/activity_create_category_flow.xml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            android:layout_marginTop="16dp"
+            android:layout_marginLeft="16dp"
+            />
+
+        <io.forus.me.android.presentation.view.component.text.TextView
+            android:id="@+id/title"
+            style="@style/TopHeader"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Choose Type"
```
Comment (maxvisser) [inline presentation/src/main/res/layout/activity_create_category_flow.xml:45]: Could you add this to strings? Dutch translation is 'Kies type'

File: presentation/src/main/res/layout/activity_create_category_flow.xml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            android:layout_below="@+id/title"
+            android:layout_marginLeft="22dp"
+            android:layout_marginRight="22dp"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="6dp"
+            android:gravity="center"
+            android:paddingEnd="27dp"
+            android:visibility="visible"
+            android:padding="8dp"
+            android:text="Select the category under which the new record type should be added"
```
Comment (maxvisser) [inline presentation/src/main/res/layout/activity_create_category_flow.xml:64]: Could you add this to strings? Dutch translation is 'Wat voor type eigenschap wilt u aanmaken?'

Comment (maxvisser) [reply presentation/src/main/res/layout/activity_create_category_flow.xml:64]: I think the english text is also incorrect. A type is not a category @jknnr

File: presentation/src/main/res/layout/activity_create_category_flow.xml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            android:layout_marginTop="10dp"
+            android:src="@drawable/ic_close_black_24dp"
+            android:tint="@color/gray_subtitle"
+            android:scaleType="center"
+            />
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Nieuwe Eigenschappen"
```
Comment (maxvisser) [inline presentation/src/main/res/layout/activity_create_category_flow.xml:35]: Text is hardcoded; could you add this to strings 'Nieuwe eigenschap' is correct dutch text. 'New record' is english

File: presentation/src/main/res/values-nl/strings.xml [hunk 4]
```diff
@@ -140,6 +141,19 @@
     <string name="record_details_validation_approved">Validatie goedgekeurd</string>
     <string name="record_details_validation_declined">Validatie geweigerd</string>
     <string name="choose_validator">Kies validator</string>
+    <string name="validations">Validaties</string>
+    <string name="validations_empty">U heeft nog geen validaties</string>
```
Comment (maxvisser) [inline presentation/src/main/res/values-nl/strings.xml:145]: Deze eigenschap heeft geen validaties

File: presentation/src/main/res/layout/view_edit_record_dialog.xml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    <TextView
+        android:id="@+id/title"
+        style="@style/Header6"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginEnd="32dp"
+        android:layout_marginTop="22dp"
+        android:fontFamily="sans-serif-medium"
+        android:text="Would you like to archive record? "
```
Comment (maxvisser) [inline presentation/src/main/res/layout/view_edit_record_dialog.xml:17]: This text is in strings but is not used.

**Comment by maxvisser** [comment]
@jknnr I would hide the edit button on this PR for now. As we discussed its a bit weird.

**Comment by maxvisser** [comment]
Some UX things I will create seperate issues for:
<details>
<summary>Submit button on creating a record is below the keyboard which is pretty annoying</summary>
image
</details>
<details>
<summary>Description of QR-code is not visible right away on large screen</summary>
image
</details>

File: data/src/main/java/io/forus/me/android/data/repository/records/RecordsRepository.kt [hunk 6]
```diff
@@ -137,6 +139,47 @@ class RecordsRepository(private val recordsRemoteDataSource: RecordsDataSource)
         }
     }
 
+    override fun deleteRecord(id: Long) : Observable<Boolean>{
+        return recordsRemoteDataSource.deleteRecord(id).map {
+           it.success
+        }
+    }
+
+    /*override fun updateRecord(id: Long, updateRecord: UpdateRecord): Observable<Success> {
```
Comment (maxvisser) [inline data/src/main/java/io/forus/me/android/data/repository/records/RecordsRepository.kt:148]: Can't this whole function be removed? (is there a reason to keep it?)

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/main/MainActivity.kt [hunk 7]
```diff
@@ -182,7 +182,7 @@ class MainActivity : BaseActivity() {
 
     private fun processInAppUpdateError(error: String) {
 
-        Toast.makeText(this@MainActivity, "initInAppUpdate error $error", Toast.LENGTH_LONG).show()
+        //Toast.makeText(this@MainActivity, "initInAppUpdate error $error", Toast.LENGTH_LONG).show()
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/main/MainActivity.kt:185]: Could you add the word 'todo' when you comment things out? So you can search later to remove all todo's if needed

File: presentation/src/main/res/layout/activity_create_category_flow.xml (hunk 2 above)
Comment (jknnr) [reply presentation/src/main/res/layout/activity_create_category_flow.xml:64]: Select the type of a new record

**Comment by maxvisser** [comment]
The back button on profile page is still there after its switched around

**Comment by maxvisser** [comment]
Archive button does nothing.

**Comment by maxvisser** [comment]
On sign-up profile icon is still visible. If I click it it will return me to my mail app. Should be removed.

File: presentation/src/main/res/values-nl/strings.xml [hunk 8]
```diff
@@ -140,6 +144,23 @@
...
+
+    <string name="dialog_record_edit_description">Alle validaties worden verwijderd. U dient opnieuw validaties aan te vragen voor de nieuwe eigenschap.</string>
+    <string name="dialog_record_archive_description">Alle validaties worden verwijderd.</string>
+
+    <string name="dialog_button_cancel">ANNULEER</string>
+    <string name="dialog_button_edit">BEWERK</string>
+    <string name="dialog_button_archive">VERWIJDER</string>
+
+    <string name="tab_records">Mijn eigenschappen</string>
+    <string name="tab_archives">Archive</string>
```
Comment (maxvisser) [inline presentation/src/main/res/values-nl/strings.xml:161]: Archief

Comment (maxvisser) [reply presentation/src/main/res/values-nl/strings.xml:161]: please change to 'Archief'

**Pull request merged**